mxcubeweb/core/components/harvester.py

```diff
@@ -0,0 +1,505 @@
+                "centring_vertical": (
+                    motor_pos_dict["centring_vertical"]
+                    - saved_position["centring_vertical"]
+                ),
+            }
+
+            calibrated_motor_offset = {
+                "focus": new_motor_offset["focus"] + new_motor_offset["centring_focus"],
+                "phiy": new_motor_offset["phiy"],
+                "phiz": (
+                    new_motor_offset["phiz"] + new_motor_offset["centring_vertical"]
+                ),
+            }
+
+            print(calibrated_motor_offset)
+            # we store the motor offset in the Harvester, to be used for sample centering
+            harvester_device.store_calibrated_pin(
+                calibrated_motor_offset["focus"],
+                calibrated_motor_offset["phiy"],
+                calibrated_motor_offset["phiz"],
+            )
+
+            harvester_device.set_calibrate_state(False)
+        except Exception as ex:
+            logging.getLogger("user_level_log").exception(
+                f"Pin Calibration / validation Failed {str(ex)}"
+            )
+            return False
+
+        return True
+
+    def get_sample_info(self, location: str) -> dict[str]:
+        samples_list = HWR.beamline.sample_changer.get_sample_list()
+        sample_data = {}
+        for s in samples_list:
+            if s.get_address() == location or s.get_id() == location:
+                sample_data = {
+                    "location": s.get_address(),
+                    "sampleID": s.get_address(),
+                    "crystalUUID": s.get_id(),
+                    "sampleName": s.get_name(),
+                }
+                return sample_data
+
+        return sample_data
+
+    def harvest_and_mount_sample(self, xtal_uuid: str) -> dict[str, bool, int]:
+        try:
+            harvester_device = HWR.beamline.harvester
+            self.harvest_crystal(xtal_uuid)
+            harvester_device._wait_sample_transfer_ready(None)
+
+            sample = self.get_sample_info(xtal_uuid)
+            self.app.sample_changer.mount_sample(sample)
+        except Exception:
+            return "Could not Harvest Crystal"
+
+        self.init_signals()
+        return self.get_harvester_contents()
+
+    def harvest_sample_before_mount(
+        self, sample_uuid: str, wait_before_load: bool = False
+    ) -> bool:
+        """
+        Return (bool): whether the sample has been harvest then mount -> (True)
+        or had and exception -> (False)
+        """
+        harvester_device = HWR.beamline.harvester
+
+        if harvester_device:
+            return harvester_device.harvest_sample_before_mount(
+                sample_uuid, wait_before_load
+            )
+        else:
+            msg = harvester_device.get_status()
+            logging.getLogger("user_level_log").exception(
+                "ERROR: No Harvester Device Found"
+            )
+            logging.getLogger("user_level_log").exception(msg)
```

Comment:
There is no exception here, we are not in an `except` block. So we should use the `error` method.

> `exception(msg, *args, **kwargs)`
>    Logs a message with level [ERROR](https://docs.python.org/3/library/logging.html#logging.ERROR) on this logger. The arguments are interpreted as for [debug()](https://docs.python.org/3/library/logging.html#logging.debug). Exception info is added to the logging message. This method should only be called from an exception handler.

-- https://docs.python.org/3/library/logging.html#logging.Logger.exception